src/Backups/BackupImpl.cpp

```diff
@@ -929,6 +935,12 @@ void BackupImpl::writeFile(const BackupFileInfo & info, BackupEntryPtr entry)
         writer->copyDataToFile(info.data_file_name, create_read_buffer, info.base_size, info.size - info.base_size);
     }
 
+    if (!deduplicate_files)
```

Comment:
The condition is not necessary because we assign `reference_sources` only if we're going to copy corresponding files. The fewer conditions the better.